src/components/form/HdRange.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div class='range'>
+    <input type="range"
+      :name="name"
+      :v-model="value"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="updateRangeDecoration"
+    >
+    <div class="range__thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      refault: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+  },
+  watch: {
+    value() {
+      this.input.value = this.value;
+      this.updateRangeDecoration();
+    },
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    this.input = this.$el.querySelector('input');
+    this.progress = this.$el.querySelector('.range__progress');
+    this.progress.style.width = '1px';
+    this.thumb = this.$el.querySelector('.range__thumb');
```

Comment:
I think it's better to use [$refs](https://vuejs.org/v2/api/#ref) instead of `this.$el.querySelector`

Reply:
Agree

Reply:
From the docs I thought it should be used just for components, but I see it's not. I go for it